models/models.go

```diff
@@ -0,0 +1,72 @@
+package models
```

Comment:
With the current problem statement i don't see any clear indicators that would promote the use of SQL over NoSQL database or vice versa. With this in mind, i would probably go with a NoSQL database (e.g.MongoDB) to allow for schema flexibility (in regards to assets) and scalability. Optimally, i would use an in-memory db (like memcached or mongodb as in-memory db) and periodically dump it to disk storage. In this way we can have the best performance. It opens up possible issues that can occur because of its volatility but there are ways around them (periodic dumping to disk, enhanced use of cache for changes, etc...).

Reply:
I see. Regarding SQL dialects, more recently i have used PL/Pg through the use of PostgreSQL. Since most of the applications that utilized this database were applications that collected/analyzed/vectorized metrics from a large amount of devices/sensors/probes/etc so they inlcuded a very large amount of operations we wanted to optimize performance as much as possible wherever possible. This was the main the reason for opting to go with raw SQL queries instead of ORMs. Not that i am against them but i prefer the versatility and the optimization opportunities that the raw queries bring to the table. Regarding patterns, nothing too overcomplicated, typical 1:1 and 1:N relationships.